From b6546177d94c87e179845216f85d81eeeba6ebe8 Mon Sep 17 00:00:00 2001
From: Maxime Petazzoni <max@signalfx.com>
Date: Thu, 10 Dec 2015 23:16:57 +0000
Subject: [PATCH] fix 0.9 build issues

---
 build.gradle |  8 ++++----
 gradlew      | 10 +++-------
 2 files changed, 7 insertions(+), 11 deletions(-)

diff --git a/build.gradle b/build.gradle
index a259cb8..da0dbe2 100644
--- a/build.gradle
+++ b/build.gradle
@@ -306,19 +306,19 @@ project(':core') {
   tasks.create(name: "genProducerConfigDocs", dependsOn:jar, type: JavaExec) {
     classpath = sourceSets.main.runtimeClasspath
     main = 'org.apache.kafka.clients.producer.ProducerConfig'
-    standardOutput = new File('docs/producer_config.html').newOutputStream()
+    standardOutput = new File("$rootDir/docs/producer_config.html").newOutputStream()
   }
 
   tasks.create(name: "genConsumerConfigDocs", dependsOn:jar, type: JavaExec) {
     classpath = sourceSets.main.runtimeClasspath
     main = 'org.apache.kafka.clients.consumer.ConsumerConfig'
-    standardOutput = new File('docs/consumer_config.html').newOutputStream()
+    standardOutput = new File("$rootDir/docs/consumer_config.html").newOutputStream()
   }
 
   tasks.create(name: "genKafkaConfigDocs", dependsOn:jar, type: JavaExec) {
     classpath = sourceSets.main.runtimeClasspath
     main = 'kafka.server.KafkaConfig'
-    standardOutput = new File('docs/kafka_config.html').newOutputStream()
+    standardOutput = new File("$rootDir/docs/kafka_config.html").newOutputStream()
   }
 
   task siteDocsTar(dependsOn: ['genProducerConfigDocs', 'genConsumerConfigDocs', 'genKafkaConfigDocs', ':connect:runtime:genConnectConfigDocs'], type: Tar) {
@@ -767,7 +767,7 @@ project(':connect:runtime') {
   tasks.create(name: "genConnectConfigDocs", dependsOn:jar, type: JavaExec) {
     classpath = sourceSets.main.runtimeClasspath
     main = 'org.apache.kafka.connect.runtime.distributed.DistributedConfig'
-    standardOutput = new File('docs/connect_config.html').newOutputStream()
+    standardOutput = new File("$rootDir/docs/connect_config.html").newOutputStream()
   }
 }
 
diff --git a/gradlew b/gradlew
index 91a7e26..9d82f78 100755
--- a/gradlew
+++ b/gradlew
@@ -42,11 +42,6 @@ case "`uname`" in
     ;;
 esac
 
-# For Cygwin, ensure paths are in UNIX format before anything is touched.
-if $cygwin ; then
-    [ -n "$JAVA_HOME" ] && JAVA_HOME=`cygpath --unix "$JAVA_HOME"`
-fi
-
 # Attempt to set APP_HOME
 # Resolve links: $0 may be a link
 PRG="$0"
@@ -61,9 +56,9 @@ while [ -h "$PRG" ] ; do
     fi
 done
 SAVED="`pwd`"
-cd "`dirname \"$PRG\"`/" >&-
+cd "`dirname \"$PRG\"`/" >/dev/null
 APP_HOME="`pwd -P`"
-cd "$SAVED" >&-
+cd "$SAVED" >/dev/null
 
 CLASSPATH=$APP_HOME/gradle/wrapper/gradle-wrapper.jar
 
@@ -114,6 +109,7 @@ fi
 if $cygwin ; then
     APP_HOME=`cygpath --path --mixed "$APP_HOME"`
     CLASSPATH=`cygpath --path --mixed "$CLASSPATH"`
+    JAVACMD=`cygpath --unix "$JAVACMD"`
 
     # We build the pattern for arguments to be converted via cygpath
     ROOTDIRSRAW=`find -L / -maxdepth 1 -mindepth 1 -type d 2>/dev/null`
-- 
2.4.1

